### ISSUE TITLE
unreachable_code_linter carve-out for 'nolint end' should also apply to 'nocov end'

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2327]
This is currently linted:

https://github.com/r-lib/lintr/blob/0815b2a1e70d8b8df47850d0309a18f0bb145329/R/get_source_expressions.R#L281

**Comment by AshesITR** [comment]
Maybe allow comments in general?

**Comment by MichaelChirico** [comment]
I guess optionally, but they're specifically included:

https://github.com/r-lib/lintr/blob/ef82966a21bf5118ec2828d58ca02287dfb02074/R/unreachable_code_linter.R#L112-L116

Often come across things like

```r
return(x)
# debugging
# print(x)
```

`commented_code_linter()` helps a little bit here but TBH that linter is far from a place where I'd apply it to my own code or to code at scale.

**Comment by AshesITR** [comment]
I see. Is there a similar way to find what `# nocov end` is configured to be?

**Comment by MichaelChirico** [comment]
Looks like we can simply read `options()`:

https://github.com/r-lib/covr/blob/54de614f5c17a42bfaef2c87dbf5723ca701326a/R/exclusions.R#L55-L57

**Comment by AshesITR** [comment]
> Looks like we can simply read `options()`:
> 
> 
> 
> https://github.com/r-lib/covr/blob/54de614f5c17a42bfaef2c87dbf5723ca701326a/R/exclusions.R#L55-L57

Sweet, and it's a regex too.
Might extend the linter to allow for a comment regex whitelist later? We can default it to nolint end and nocov end.
